functions/src/utils.ts

```diff
@@ -103,3 +100,41 @@ export function isValidUrl(urlString: string | undefined): boolean {
   }
   return url.protocol === "http:" || url.protocol === "https:";
 }
+
+/** Verify a Firebase ID token is valid.
+ *
+ * @param token Firebase ID token to verify.
+ * @returns True if the token is valid, throws an error otherwise.
+ */
+export async function verifyIdToken(token: string | undefined) {
+  if (!token) {
+    throw new ShareLinkError(ShareLinkErrorCode.INVALID_TOKEN);
+  }
+  return admin
+    .auth()
+    .verifyIdToken(token)
+    .then(() => true)
+    .catch((error) => {
+      console.error(error);
+      throw new ShareLinkError(ShareLinkErrorCode.INVALID_TOKEN);
+    });
+}
+
+/** Coerce a value to a boolean or throw an error if not possible.
+ *
+ * @param value The value to coerce.
+ * @returns The coerced boolean value.
+ */
+export function parseBoolean(value: unknown): boolean {
+  if (typeof value === "boolean") return value;
+  if (typeof value === "number" && [0, 1].includes(value)) {
+    return value === 1;
+  }
+  if (
+    typeof value === "string" &&
+    (value.toLowerCase() === "true" || value.toLowerCase() === "false")
+  ) {
+    return value.toLowerCase() === "true";
+  }
+  throw new Error("Invalid boolean value");
```

Comment:
TBH I think I'd rather be strict and just require them to pass "true" or "false".  Else there could be confusion like `1` works but `"1"` doesn't, etc.  And it gets more complicated than necessary.

Reply:
makes sense--updated to just use booleans!